thirdparty/pimlico.go

```diff
@@ -209,9 +212,17 @@ func (v *PimlicoVendor) OwnsUpstream(ups *common.UpstreamConfig) bool {
 }
 
 func (v *PimlicoVendor) createClient(ctx context.Context, logger *zerolog.Logger, parsedURL *url.URL) (clients.HttpJsonRpcClient, error) {
-	client, err := clients.NewGenericHttpJsonRpcClient(ctx, logger, "n/a", "n/a", parsedURL, nil)
-	if err != nil {
-		return nil, err
+	if v.proxyPoolRegistry != nil {
+		proxyPool, err := v.proxyPoolRegistry.GetPool(parsedURL.Host)
+		if err != nil {
+			return nil, err
+		}
+		client, err := clients.NewGenericHttpJsonRpcClient(ctx, logger, "n/a", "n/a", parsedURL, nil, proxyPool)
```

Comment:
We don't need to pass proxyPoolRegistry to vendors at all, none of these clients are used for critical traffic. These are mainly one-off used to detect chainId.

If these vendors needed proxy in future we can make sure to pass the ProxyPool object maybe as part of VendorSettings.

For now I suggest to remove them.